components/AddressInput.js

```diff
@@ -0,0 +1,143 @@
+import { useState, useEffect, useRef } from 'react'
+import Select from 'react-select'
+import { useTranslation } from 'next-i18next'
+import { useRouter } from 'next/router'
+
+import {
+  isValidCTID,
+} from '../utils'
+
+import { userOrServiceLink } from '../utils/format'
+
+let typingTimer
+
+export default function AddressInput({ searchPlaceholderText, userData = {}, setFilters, type, inputValue, title, link }) {
+  const { t } = useTranslation()
+  const router = useRouter()
+  const searchInput = useRef(null)
+
+  const { id } = router.query
+  const [notEmpty, setNotEmpty] = useState(false)
+  const [searchItem, setSearchItem] = useState(id || userData?.address || inputValue || "")
+  const [errorMessage, setErrorMessage] = useState("")
+  const [isMounted, setIsMounted] = useState(false)
+
+  useEffect(() => {
+    setIsMounted(true)
+  }, [])
+
+  useEffect(() => {
+    if (userData?.address) {
+      setSearchItem(userData.address)
+    }
+  }, [userData])
+
+
+  const searchOnKeyUp = e => {
+    const value= e?.target?.value;
+
+    if (isValidCTID(value)) {
+      return
+    }
+
+    //if more than 3 characters - search for suggestions
+    let notEmpty = false;
+    if (value && value.length > 0) {
+      clearTimeout(typingTimer)
+      typingTimer = setTimeout(async () => {
+
+        if(type === 'search' || type === 'issuer') {
```

Comment:
What are the differences between a search and an issuer?

Reply:
'search' is a type of name input, I changed to the 'name' to make it more clear

Reply:
then issues isn't very clear either:) 

user can type in
a username 
an address
a xaman slug
a globaild slug
a service name 

if an address - then it can be `address` - if not an address (all the rest) it can be `name`

Reply:
I removed this part of the code to make it clearer